cmd/dfget/cmd/root.go

```diff
@@ -201,24 +201,21 @@ func checkAndSpawnDaemon() (client.DaemonClient, error) {
 		return nil, err
 	}
 
-	// 3.First check since starting
-	if daemonClient.CheckHealth(context.Background(), target) == nil {
-		return daemonClient, nil
-	}
-
-	times := 0
-	limit := 100
-	interval := 50 * time.Millisecond
-	for {
-		// 4.Cycle check with 5s timeout
-		if daemonClient.CheckHealth(context.Background(), target) == nil {
-			return daemonClient, nil
+	// 3. check health with 5s timeout
+	var ch = make(chan int, 1)
+	go func() {
+		for {
+			if err = daemonClient.CheckHealth(context.Background(), target); err == nil {
+				ch <- 1
+				break
+			}
 		}
```

Comment:
```golang

for {		
        if err = daemonClient.CheckHealth(context.Background(), target); err != nil  {	
	    continue				
	}
        ch <- 1
        break
}

```

Reply:
I don't see the meaning of this early return. But you do remind me I should exit the goroutine when timeout happens.

Reply:
early return? what is mean?

Reply:
Early return means return the code block early. I update commit and borrowed your code, check it.